src/azure-cli/azure/cli/command_modules/redis/custom.py

```diff
@@ -177,4 +184,84 @@ def cli_redis_force_reboot(client, resource_group_name, name, reboot_type, shard
     param = RedisRebootParameters(reboot_type=reboot_type, shard_id=shard_id)
     return client.force_reboot(resource_group_name, name, param)
 
+
+def cli_redis_identity_show(client, resource_group_name, cache_name):
+    from azure.mgmt.redis.models import ManagedServiceIdentityType, ManagedServiceIdentity
+    redis_resourse = client.get(resource_group_name, cache_name)
+    if redis_resourse.identity is None:
+        redis_resourse.identity = ManagedServiceIdentity(
+            type=ManagedServiceIdentityType.NONE.value,
+        )
+    return redis_resourse.identity
+
+
+def cli_redis_identity_assign(client, resource_group_name, cache_name, mi_system_assigned=None, mi_user_assigned=None):
+    from azure.mgmt.redis.models import RedisUpdateParameters, ManagedServiceIdentityType
+    redis_resourse = client.get(resource_group_name, cache_name)
+    identity = redis_resourse.identity
+    if identity is not None:
+        if ManagedServiceIdentityType.SYSTEM_ASSIGNED.value in identity.type:
+            mi_system_assigned = True
+        if ManagedServiceIdentityType.USER_ASSIGNED.value in identity.type:
+            old_user_identity = list(identity.user_assigned_identities)
+            if mi_user_assigned is None:
+                mi_user_assigned = []
+            for user_id in old_user_identity:
+                mi_user_assigned.append(user_id)
+    update_params = RedisUpdateParameters(
+        identity=build_identity(mi_system_assigned, mi_user_assigned))
+    redis_resourse = client.update(resource_group_name, cache_name, update_params)
+    return redis_resourse.identity
```

Comment:
Could you use the unified method [`azure.cli.core.commands.arm.assign_identity`](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/commands/arm.py#L712) assign identities? Please refer to [the code sample](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/resource/custom.py#L2475) for details

Reply:
@zhoxing-ms I don't think it is compatible with our redis module since we don't support roles and scopes.

Reply:
OK